readthedocs/search/api/v2/serializers.py

```diff
@@ -129,10 +135,15 @@ def _get_project_data(self, obj):
         if project_data:
             return project_data
 
-        project = Project.objects.filter(slug=obj.project).first()
-        if project:
+        version = (
+            Version.objects.filter(project__slug=obj.project, slug=obj.version)
+            .select_related("project")
+            .first()
+        )
```

Comment:
Instead of querying the project and version separately, we can do it in one query.